### PR TITLE
add howtorun.md to perf-tests/clusterloader2

### PR DESCRIPTION
**Pull request by sonyafenge**
**What type of PR is this?**

> /kind documentation

**What this PR does / why we need it**:
add instructions and required enviroments/args for perf-tests
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: perf-tests/clusterloader2/howtorun.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+Process under GCP project. Please remember perf-tests running on kubemark master, so all logs should be checked under /var/log/ on kubemark-master
+
+## Pre-rerequisites: GCP config
+1. run "gcloud version" to ensure your Google Cloud SDK is updated (suggested Google Cloud SDK 298.0.0 and up), Please refer to https://cloud.google.com/sdk/docs/downloads-apt-get or https://cloud.google.com/sdk/docs/downloads-versioned-archives to upgrade your google cloud SDK
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/howtorun.md:4]: first letter in upper case. applies to all the other paragraphs.

File: perf-tests/clusterloader2/howtorun.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+2. ensure your docker login and access has been configured, if not, please run "gcloud auth configure-docker" to config
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/howtorun.md:5]: how? maybe give a docker command. "ensure docker has been configured" is a bit vague.

File: perf-tests/clusterloader2/howtorun.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+5. after all run finished, shutdown cluster
+```
+$ export MASTER_ROOT_DISK_SIZE=100GB MASTER_DISK_SIZE=200GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-8 NUM_NODES=8 NODE_DISK_SIZE=200GB KUBE_GCE_NETWORK=kubemark-500 GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=kubemark-500 ENABLE_KCM_LEADER_ELECT=false
+$ ./test/kubemark/stop-kubemark.sh 
+$ ./cluster/kube-down.sh
+```
+
+## PartitionServer Config
+Arktos support multi data partition server including apiserver, workload-controller-managere, ETCD. if you want to start multi data partition server, Please add the env below
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/howtorun.md:49]: env -> env variables

File: perf-tests/clusterloader2/howtorun.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+### APISERVERS_EXTRA_NUM
+set extra apiserver number, default equals 0.  
+if you have extra apiserver, please ensure "--api-server-addresses=https://${extra apiserver external IP}" is added when you run perf-tests. Split by ; for multi servers.
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/howtorun.md:52]: extra apiserver -> an extra apiserver

Comment (pdgetrf) [reply perf-tests/clusterloader2/howtorun.md:52]: Split by -> separated by

File: perf-tests/clusterloader2/howtorun.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+### WORKLOADCONTROLLER_EXTRA_NUM
+set extra workload controller member number, , default equals 0
```
Comment (pdgetrf) [inline perf-tests/clusterloader2/howtorun.md:54]: extra comma

**Comment by pdgetrf** [comment]
the title of the document should indicate it's about perf test with multiple apiserver. "howtorun" tells too little information.

**Comment by Sindica** [comment]
Suggest separate the manual to a few sections: General Setup (here we can have quick start), Testing Scenarios (here you can explain additional environment variables and arguments), And an summary for all env variables at the end. Thanks.

**Comment by sonyafenge** [comment]
@pdgetrf This doc is under clusterloader2 folder and only for clusterloader. In community, perf-tests including bunch of tools, clusterloader is only one of the tools. use "howtorun" as title is to match with "README" under same folder.

**Comment by pdgetrf** [comment]
> @pdgetrf This doc is under clusterloader2 folder and only for clusterloader. In community, perf-tests including bunch of tools, clusterloader is only one of the tools. use "howtorun" as title is to match with "README" under same folder.

none of these seems relevant to me as to why this file is named "howtorun". I don't even understand why we have a folder called "cluserloader2". where's the 2 come from? what does 2 mean? same applies to "howtorun". A filename should be self-explanatory. If our user come into this folder and saw this name, will he have any idea what is to be run by seeing his filename?

**Comment by sonyafenge** [comment]
> > @pdgetrf This doc is under clusterloader2 folder and only for clusterloader. In community, perf-tests including bunch of tools, clusterloader is only one of the tools. use "howtorun" as title is to match with "README" under same folder.
> 
> none of these seems relevant to me as to why this file is named "howtorun". I don't even understand why we have a folder called "cluserloader2". where's the 2 come from? what does 2 mean? same applies to "howtorun". A filename should be self-explanatory. If our user come into this folder and saw this name, will he have any idea what is to be run by seeing his filename?

Perf-tests/clusterloader2 is the name from community perf-tests repo, it is one of the community perf-test tools: https://github.com/kubernetes/perf-tests/tree/master/clusterloader2

File: perf-tests/clusterloader2/howtorun.md (hunk 1 above)
Comment (sonyafenge) [reply perf-tests/clusterloader2/howtorun.md]: updated all paragraphs first letter to upper case

File: perf-tests/clusterloader2/howtorun.md (hunk 2 above)
Comment (sonyafenge) [reply perf-tests/clusterloader2/howtorun.md]: updated docker command.

File: perf-tests/clusterloader2/howtorun.md (hunk 3 above)
Comment (sonyafenge) [reply perf-tests/clusterloader2/howtorun.md]: updated

File: perf-tests/clusterloader2/howtorun.md (hunk 4 above)
Comment (sonyafenge) [reply perf-tests/clusterloader2/howtorun.md]: updated to "extra apiservers" per there may be multi.
updated to "separated by"

File: perf-tests/clusterloader2/howtorun.md (hunk 5 above)
Comment (sonyafenge) [reply perf-tests/clusterloader2/howtorun.md]: fixed

**Comment by pdgetrf** [comment]
> > > @pdgetrf This doc is under clusterloader2 folder and only for clusterloader. In community, perf-tests including bunch of tools, clusterloader is only one of the tools. use "howtorun" as title is to match with "README" under same folder.
> > 
> > 
> > none of these seems relevant to me as to why this file is named "howtorun". I don't even understand why we have a folder called "cluserloader2". where's the 2 come from? what does 2 mean? same applies to "howtorun". A filename should be self-explanatory. If our user come into this folder and saw this name, will he have any idea what is to be run by seeing his filename?
> 
> Perf-tests/clusterloader2 is the name from community perf-tests repo, it is one of the community perf-test tools: https://github.com/kubernetes/perf-tests/tree/master/clusterloader2

take a look inside clusterload2 folder

image

there are two scripts run-e2e.sh and clean-up-old-snapshots.sh and these names all at least give info on what is being run. having a doc called howtorun.md is like having a script called run.sh or dostuff.sh. It doesn't tell enough info by the look. Those who have never used it before will not know what it does.

**Comment by sonyafenge** [comment]
Close this Pr per scenarios has been changed

**Pull request closed without merging**